src/backend/distributed/commands/alter_table.c

```diff
@@ -1138,7 +1138,7 @@ ReplaceTable(Oid sourceId, Oid targetId, List *justBeforeDropCommands,
 	{
 		changeDependencyFor(RelationRelationId, sequenceOid,
 							RelationRelationId, sourceId, targetId);
-		if (ShouldSyncTableMetadata(sourceId))
+		if (IsCitusTable(sourceId) && ShouldSyncTableMetadata(sourceId))
```

Comment:
it seems slightly better to move `IsCitusTable(sourceId)` into `ShouldSyncTableMetadata ` and return false?

Like:
```

if (relationId == InvalidOid || !IsCitusTable(relationId))
{

return false;
}